tensilelite/rocisa/cmake/RocisaConfig.cmake

```diff
@@ -0,0 +1,74 @@
+#
+# Copyright (C) 2022-2025 Advanced Micro Devices, Inc. All rights reserved.
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+#
+################################################################################
+
+include(CMakeParseArguments)
+if(NOT DEFINED Rocisa_ROOT)
+    # Compute the installation prefix relative to this file.
+    get_filename_component(Rocisa_PREFIX "${CMAKE_CURRENT_LIST_FILE}" PATH)
+    get_filename_component(Rocisa_PREFIX "${Rocisa_PREFIX}" PATH)
+    if (WIN32)
+        execute_process(COMMAND "${Rocisa_PREFIX}/bin/RocisaGetPath.exe" OUTPUT_VARIABLE Rocisa_ROOT)
+    else()
+        execute_process(COMMAND "${Rocisa_PREFIX}/bin/RocisaGetPath" OUTPUT_VARIABLE Rocisa_ROOT)
+    endif()
+endif()
+
+message(STATUS "Building rocIsa" ${Rocisa_ROOT})
+
+set(ROCISA_VIRTUAL_DIR ${Rocisa_ROOT})
+file(MAKE_DIRECTORY ${ROCISA_VIRTUAL_DIR}/build)
+execute_process(
+  COMMAND ${CMAKE_COMMAND} -DPython_EXECUTABLE=${VIRTUALENV_BIN_DIR}/${VIRTUALENV_PYTHON_EXENAME} ..
+  WORKING_DIRECTORY ${ROCISA_VIRTUAL_DIR}/build
+  OUTPUT_VARIABLE output
+  ERROR_VARIABLE error
+  RESULT_VARIABLE result
+)
+if(NOT ${result} EQUAL 0)
+  message(STATUS "Output: ${output}")
+  message(STATUS "Error: ${error}")
+  message(STATUS "Result: ${result}")
+else()
+  message(STATUS "rocIsa configuring done.")
+endif()
+execute_process(
+  COMMAND nproc
+  OUTPUT_VARIABLE NPROC_COUNT
+  OUTPUT_STRIP_TRAILING_WHITESPACE
+)
+execute_process(
+  COMMAND ${CMAKE_COMMAND} --build . -j ${NPROC_COUNT}
+  WORKING_DIRECTORY ${ROCISA_VIRTUAL_DIR}/build
+  OUTPUT_VARIABLE output
+  ERROR_VARIABLE error
+  RESULT_VARIABLE result
+)
+if(NOT ${result} EQUAL 0)
+  message(STATUS "Output: ${output}")
+  message(STATUS "Error: ${error}")
+  message(FATAL_ERROR "Result: ${result}")
+else()
+  message(STATUS "rocIsa building done.")
+  set(ENV{PYTHONPATH} "${ROCISA_VIRTUAL_DIR}/build/lib:$ENV{PYTHONPATH}")
```

Comment:
A note on what is consuming this PYTHONPATH please. It is very irregular to be communicating by env var within a CMake configure session like this.